Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js

```diff
@@ -1625,15 +1625,24 @@ WI.SourcesNavigationSidebarPanel = class SourcesNavigationSidebarPanel extends W
             return true;
         }
 
-        case WI.DebuggerManager.PauseReason.Assertion:
+        case WI.DebuggerManager.PauseReason.Assertion: {
             // FIXME: We should include the assertion condition string.
             console.assert(pauseData, "Expected data with an assertion, but found none.");
-            if (pauseData && pauseData.message)
-                this._pauseReasonTextRow.text = WI.UIString("Assertion with message: %s").format(pauseData.message);
-            else
-                this._pauseReasonTextRow.text = WI.UIString("Assertion Failed");
-            this._pauseReasonGroup.rows = [this._pauseReasonTextRow];
+
+            this._pauseReasonTreeOutline = this.createContentTreeOutline({suppressFiltering: true});
+
+            let assertionFailuresBreakpointTreeElement = new WI.JavaScriptBreakpointTreeElement(WI.debuggerManager.assertionFailuresBreakpoint, {
+                classNames: ["paused"],
+                title: pauseData?.message ? WI.UIString("Assertion with message: %s").format(pauseData.message) : WI.UIString("Assertion Failed"),
+            });
+            this._pauseReasonTreeOutline.appendChild(assertionFailuresBreakpointTreeElement);
+
+            let assertionFailuresBreakpointRow = new WI.DetailsSectionRow;
+            assertionFailuresBreakpointRow.element.appendChild(this._pauseReasonTreeOutline.element);
+
+            this._pauseReasonGroup.rows = [assertionFailuresBreakpointRow];
```

Comment:
Can we generalize this into a helper function that takes the `*TreeElement` and then sets to pause reason rows to reduce the duplication this patch is doing?

Reply:
i was originally thinking of doing this as a "give me the breakpoint and title and I'll create the tree element" but hesitated due to the difference in how each tree element was created

i like your idea better :)